src/transaction/log_page_buffer.c

```diff
@@ -3988,12 +3989,22 @@ logpb_flush_all_append_pages (THREAD_ENTRY * thread_p)
 	  error_code = ER_FAILED;
 	  goto error;
 	}
-      tmp_eof = (LOG_RECORD_HEADER *) ((char *) first_append_log_page->area + log_Gl.append.prev_lsa.offset);
+
+      copy_to_first_append = (LOG_PAGE *) malloc (LOG_PAGESIZE);
```

Comment:
@funkydvd 

Comments describe what we are going to do will greatly help other developers to understand your intention.